tools/m2kcli/commands/context/context.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019 Analog Devices Inc.
+ *
+ * This file is part of libm2k
+ * (see http://www.github.com/analogdevicesinc/libm2k).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef M2KCLI_CONTEXT_H
+#define M2KCLI_CONTEXT_H
+
+#include <libm2k/m2k.hpp>
+#include <libm2k/context.hpp>
+#include "commands/command_in.h"
+
+class DMM;
+
+namespace libm2k {
+namespace cli {
+
+
+
+class Context : public CommandIn {
```

Comment:
I think that Context should inherit only the Command class. CommandIn is a specialization of the Command for handling  input buffers.